Update `python-dev` mailing list to Discourse in `.github/CONTRIBUTING.rst` (#138996)
python/cpython: .github/CONTRIBUTING.rst

- All discussions that are not directly related to the code in the pull request
  should happen on `GitHub Issues <https://github.com/python/cpython/issues>`_.
- Upon your first non-trivial pull request (which includes documentation changes),

>   feel free to add yourself to `Misc/ACKS <https://github.com/python/cpython/blob/main/Misc/ACKS>`_.

Reviewer: We don't need a link here

Setting Expectations
--------------------

> Due to the fact that this project is volunteer-run,

Reviewer: ```suggestion
Due to the fact that this project is run by volunteers,
```

> we unfortunately can make no guarantees as to if

Reviewer: ```suggestion
unfortunately we cannot make any guarantees as to if
```

> Due to the fact that this project is primarily volunteer-run,
> we unfortunately can make no guarantees as to if

Reviewer: I understand where you're coming from but I believe even the developers in residence have non-CPython responsibilities.

```suggestion

Due to the fact that this project is volunteer-run,
we unfortunately can make no guarantees as to if
```

or when a core developer will get around to reviewing your pull request.
If no core developer has done a review or responded to changes made because of a

> "changes requested" review, you can post on the `Core Development Discourse
> category <https://discuss.python.org/c/core-dev/23>`_ to ask for someone to
> review your pull request.

Reviewer: Also, add an explicit time period. A month without change is reasonable to ask for an update, anything less than that we don't want people pinging about.
```suggestion
'changes requested' review within a month, you can ask for someone to
review your pull request via a post in the `Core Development Discourse
category <https://discuss.python.org/c/core-dev/23>`__
```
